lib/_http_server.js

```diff
@@ -220,18 +222,22 @@ function writeHead(statusCode, reason, obj) {
   }
   this.statusCode = statusCode;
 
-  var headers;
+  headers = undefined;
   if (this[outHeadersKey]) {
     // Slow-case: when progressive API and header fields are passed.
-    var k;
-    if (obj) {
+    var key;
+    if (typeof obj === 'object') {
```

Comment:
Wouldn't this need to check for `null` as well?

Reply:
Yes, fixed for http2 compat as well.